channeldb/migrations.go

```diff
@@ -458,3 +459,99 @@ func paymentStatusesMigration(tx *bolt.Tx) error {
 
 	return nil
 }
+
+// migratePruneEdgeUpdateIndex is a database migration that attempts to resolve
+// some lingering bugs with regards to edge policies and their update index.
+// Stale entries within the edge update index were not being properly pruned due
+// to a miscalculation on the offset of an edge's policy last update. This
+// migration also fixes the case where the public keys within edge policies were
+// being serialized with an extra byte, causing an even greater error when
+// attempting to perform the offset calculation described earlier.
+func migratePruneEdgeUpdateIndex(tx *bolt.Tx) error {
```

Comment:
A remark on the side: in this PR, a migration is used to clean up invalid data in the database. I am wondering if this should also be the way to get rid of unreadable channel data (https://github.com/lightningnetwork/lnd/pull/1295). Maybe when the bug that caused the invalid data was fixed, a migration should have been added to clean the db.

Reply:
Good point, we could also extend this approach to the #1295. However, we'll need to be careful to ensure that we don't accidentally delete non-borked channel state. We can recover if we delete graph state, but channel state is _much_ more precious.